timemachine/fe/single_topology.py

```diff
@@ -63,6 +79,87 @@ def recursive_map(items, mapping):
         return mapping[items]
 
 
+def setup_dummy_bond_and_chiral_interactions_from_ff(
+    ff, mol, dummy_group, root_anchor_atom, core_atoms, chiral_atom_k, chiral_bond_k
+):
+    """
+    Setup interactions involving atoms in a given dummy group.
+    """
+    top = topology.BaseTopology(mol, ff)
+
+    bond_params, hb = top.parameterize_harmonic_bond(ff.hb_handle.params)
+    chiral_atom_potential, _ = top.setup_chiral_restraints(chiral_atom_k, chiral_bond_k)
+    chiral_atom_idxs = chiral_atom_potential.potential.idxs
+    chiral_atom_params = chiral_atom_potential.params
+
+    return setup_dummy_bond_and_chiral_interactions(
+        hb.idxs,
+        bond_params,
+        chiral_atom_idxs,
+        chiral_atom_params,
+        dummy_group,
+        root_anchor_atom,
+        core_atoms,
+    )
+
+
+def setup_dummy_bond_and_chiral_interactions(
+    bond_idxs,
+    bond_params,
+    chiral_atom_idxs,
+    chiral_atom_params,
+    dummy_group,
+    root_anchor_atom,
+    core_atoms,
+):
+    assert root_anchor_atom in core_atoms
+
+    dummy_bond_idxs = []
+    dummy_bond_params = []
+    dummy_chiral_atom_idxs = []
+    dummy_chiral_atom_params = []
+
+    # dummy_group may be a set in certain cases, so sanity check.
+
+    assert len(dummy_group) == len(list(dummy_group))
+    dummy_group = list(dummy_group)
+
+    # dummy group and anchor
+    dga = dummy_group + [root_anchor_atom]
+
+    # copy interactions that involve only root_anchor_atom
+    for idxs, params in zip(bond_idxs, bond_params):
+        if all([a in dga for a in idxs]):
+            dummy_bond_idxs.append(tuple([int(x) for x in idxs]))  # tuples are hashable etc.
+            dummy_bond_params.append(params)
+
+    # certain configuration of chiral states are symmetrizable
+    # . means a bond involving at least 1 dummy atom
+    # | means a bond involving only core atoms
+    #
+    #        all allowed geometries              |  disallowed geometry
+    #     center is core     center is not core  |
+    #    d      c      c      d      d      d    |    c      c
+    #    .      |      |      .      .      .    |    .      |
+    #    c      c      c      d      d      d    |    d      c
+    #   . .    . .    / .    . .    . .    . .   |   . .    / \
+    #  d   d  d   d  c   d  c   d  c   c  d   d  |  c   c  c   c (only core)
+    dgc = dummy_group + list(core_atoms)
+    for idxs, params in zip(chiral_atom_idxs, chiral_atom_params):
+        center, i, j, k = idxs
+        if all([a in dgc for a in idxs]):
+            # non center dummy atom count
+            ncda_count = sum([a in dummy_group for a in (i, j, k)])
+            if ncda_count == 1 or ncda_count == 2 or ncda_count == 3:
```

Comment:
Any reason not to do the following? Looks like value can only be [0, 3]
```suggestion
            if ncda_count > 0:
```

Reply:
Not sure; this code was lifted from #1358 (and even there might have just been copy-pasted). Seems reasonable to me, but I think @proteneer was the original author (maybe there's some reason to be explicit?)